text/0000-eliminate-finalize.md

```diff
@@ -0,0 +1,142 @@
+public excess: 0x03639592cc3d406a1773ead98b59aa0c54c10e6412edfcdf78c45faf6c810d90d4
+public nonce: 0x0332e9abd666ab919b22613e7f6332f449c9adb46b323b0d069953faabf15cfd45
+bech32 address: grin1qx87jxj3htwwhj5k6hy0xhd0fszpntznz26clfjfdsmpggy437gt5qmrjkfvc02qdgth86ke3dv65rz5cy8xgyhdln0h33zl4akgzrvs6spn96dt6en2hyvmyfsnulmrxt6ynjddk34nywcdq6v4874t79w063gugaa22
+
+## Payment Proofs
+
+### Building the proof
+
+Define `diffie_secret` as the Diffie-Hellman secret, which can be computed by only the sender and the receiver:
+`diffie_secret` = `Hash(sender.nonce * receiver.public_nonce)` = `Hash(sender.public_nonce * receiver.nonce)`
+
+Calculate `proof_nonce` = `Hash(diffie_secret | receiver.public_excess | tx.amount)`.
+
+The `total_nonce` for the transaction will then be `sender.nonce + receiver.nonce + proof_nonce`.
+
+* `e = SHA256(M | sender.public_nonce + receiver.public_nonce + (proof_nonce * G) | sender.public_excess + receiver.public_excess)`
+* The sender's partial signature will be `sender.partial_sig = sender.nonce + e * sender.excess`
+* The receiver's partial signature will be `receiver.partial_sig = receiver.nonce + e * receiver.excess` 
+
+The final aggregated kernel signature `(s, k*G)` can then be calculated as `(sender.partial_sig + receiver.partial_sig + proof_nonce, total_nonce * G)`
+
+### Verifying the proof
+
+The following 4 steps are necessary for the sender to prove payment to the receiver's address:
+
+1. Provide the kernel and show that it was confirmed on-chain.
+2. Prove knowledge of `sender.nonce`
+3. Show that `sender.public_nonce + receiver.public_nonce + (proof_nonce * G)` = `total_nonce * G` (ie. the `k*G` in the kernel signature)
+4. Provide the preimage to `proof_nonce` (`diffie_secret | receiver.public_excess | tx.amount`)
+
+## Slate Format
+
+A slate is now only needed for passing information from the sender to receiver. It must include the following data:
+
+* `tx.amount`
+* `tx.offset`
+* `tx.kernel_features`
+* `tx.fee`
+* `tx.inputs`
+* `receiver.change_output`
+* `receiver.public_excess` & `receiver.public_nonce` (to prevent the need for grinding)
+* `sender.public_excess`, `sender.public_nonce`, & `sender.partial_sig`
+* `time_to_live` (optional)
+
+TODO: Define JSON & binary formats
+TODO: Mention QR codes
+
+## Backward Compatibility
+
+TODO: Decide how to handle versioning. In theory, this should be simple. If given an old slatepack address, use the old slate format. If given a new, one-time address, then the new slate format is supported.
+
+## Invoices
+
+To send an invoice to a payer, the recipient provides the payer with a newly generated one-time address, an amount, and any additional fields we decide to support (memo, refund address, expiration, etc). The payer, upon approving payment of the invoice, simply follows the standard 2-step send process described in this RFC. There's no need for a special workflow.
+
+## Security
```

Comment:
What about Expiring Kernels / Unique Kernels? Is that orthogonal to this proposal? I thought I read somewhere that it was a pre-requisite, did I misunderstand that?